gulp-core-build-typescript/gulpfile.js

```diff
@@ -12,4 +14,12 @@ build.typescript.setConfig({
   typescript: require('typescript')
 });
 
-build.initialize(require('gulp'));
\ No newline at end of file
+// We need to wrap this class in a getter because it hasn't been compiled by the time this file is executed
+const taskWrapper = {};
+Object.defineProperty(taskWrapper, "default", {
+  get: () => require('./lib/RunApiExtractorOnExternalApiTypes.js').default
```

Comment:
@iclanton @nickpape-msft The idea of ExternalApiHelper.generateApiJson() was to push all the complicated logic down into a TypeScript function that could be compiled as part of api-extractor, so that the gulpfile would just be a few simple lines that invoke it.  This would eliminate the need for a weird wrapper that uses the gulpfile to compile part of its own code.

The one wrinkle is that api-extractor should not have a dependency on GulpTask<void>, so you would need to solve that somehow.  Is that doable?